lib/gc/Transforms/GPU/AllocsToSLM.cpp

```diff
@@ -80,22 +81,17 @@ struct ConvertAlloc : public OpRewritePattern<memref::AllocOp> {
       return rewriter.notifyMatchFailure(
           allocOp, "Only support constant block sizes for now");
 
-    int64_t xI = xSz.value();
-    int64_t yI = ySz.value();
-    int64_t zI = zSz.value();
-
-    if (zI != 1)
-      return rewriter.notifyMatchFailure(
-          allocOp, "Only support 2D shared memory for now");
-
+    SmallVector<int64_t, 3> blockSizes = {xSz.value(), ySz.value(),
+                                          zSz.value()};
```

Comment:
Do you really need a Vector herre? It's only used to get elements by intex and could be replaced with either local variables or int64_t[] array.

Reply:
changed vector to a static array here